src/vs/platform/terminal/common/capabilities/shellEnvDetectionCapability.ts

```diff
@@ -36,4 +37,33 @@ export class ShellEnvDetectionCapability extends Disposable implements IShellEnv
 		// Convert to event and fire event
 		this._onDidChangeEnv.fire(this._env);
 	}
+
+	startEnvironmentSingleVar(isTrusted: boolean): void {
+		if (!isTrusted) {
+			return;
+		}
+		this._env.clear();
```

Comment:
Don't need to clear anymore

Reply:
I removed them [here](https://github.com/microsoft/vscode/pull/237602/commits/532ef4161cbd57659c807a2b4709fa683f57738e ): 
but would we ever have to worry about a case where we start "startEnvironmentSingleVar" but User requests to access terminal.shellIntegration.env before we go into "setEnvironmentSingleVar" and "endEnvironmentSingleVar"?

Reply:
If that happens I would expect the old state, rather than a partially complete new state